app/components/post_list/post/body/content/permalink_preview/permalink_preview.tsx

```diff
@@ -0,0 +1,214 @@
+            backgroundColor: theme.centerChannelBg,
+            borderColor: changeOpacity(theme.centerChannelColor, 0.16),
+            borderWidth: 1,
+            borderRadius: 4,
+            marginTop: 8,
+            marginBottom: 8,
+            padding: 12,
+            shadowOffset: {
+                width: 0,
+                height: 2,
+            },
+            shadowOpacity: 0.08,
+            shadowRadius: 3,
+            elevation: 2,
+        },
+        header: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            marginBottom: 8,
+        },
+        authorInfo: {
+            flex: 1,
+            marginLeft: 12,
+            flexDirection: 'row',
+            alignItems: 'center',
+        },
+        authorName: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100, 'SemiBold'),
+        },
+        timestamp: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+            marginLeft: 8,
+        },
+        messageContainer: {
+            marginBottom: 8,
+            maxHeight: 20 * MAX_PERMALINK_PREVIEW_LINES,
+            overflow: 'hidden',
+        },
+        messageText: {
+            color: theme.centerChannelColor,
+            ...typography('Body', 100),
+            lineHeight: 20,
+        },
+
+        channelContext: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 75),
+        },
+        channelName: {
+            color: theme.linkColor,
+            ...typography('Body', 75, 'SemiBold'),
+        },
+    };
+});
+
+const PermalinkPreview = ({embedData, showPermalinkPreviews, author, locale, teammateNameDisplay, isOriginPostDeleted, location}: PermalinkPreviewProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+
+    const textStyles = getMarkdownTextStyles(theme);
+    const blockStyles = getMarkdownBlockStyles(theme);
+
+    if (!showPermalinkPreviews || isOriginPostDeleted) {
+        return null;
+    }
+
+    const {
+        post,
+        channel_display_name,
+        channel_type,
+    } = embedData;
+
+    const truncatedMessage = useMemo(() => {
+        const message = post?.message;
+
+        if (!message || typeof message !== 'string') {
+            return '';
+        }
+
+        const cleanMessage = message.trim();
+        const lines = cleanMessage.split('\n');
+        if (lines.length > MAX_PERMALINK_PREVIEW_LINES) {
+            return lines.slice(0, MAX_PERMALINK_PREVIEW_LINES).join('\n') + '...';
+        }
+
+        return cleanMessage;
+    }, [post?.message]);
+
+    const isEdited = useMemo(() => post && post.edit_at, [post]);
+
+    const authorDisplayName = useMemo(() => {
+        if (author) {
+            return displayUsername(author, locale, teammateNameDisplay);
+        }
+        return displayUsername(undefined, locale, teammateNameDisplay);
```

Comment:
no need for the if condition


```suggestion
        return displayUsername(author, locale, teammateNameDisplay);
```

Reply:
Done.